src/sentry/eventstore/snuba/backend.py

```diff
@@ -108,9 +110,62 @@ def __get_events(
         should_bind_nodes=False,
     ):
         assert filter, "You must provide a filter"
-        cols = self.__get_columns(additional_columns)
+        cols = self.__get_columns()
         orderby = orderby or DESC_ORDERING
 
+        # This is an optimization for the Group.filter_by_event_id query where we
+        # have a single event ID and want to check all accessible projects for a
+        # direct hit. In this case it's usually faster to go to nodestore first.
+        if (
+            filter.event_ids
+            and filter.project_ids
+            and len(filter.event_ids) * len(filter.project_ids) < min(limit, NODESTORE_LIMIT)
+            and offset == 0
+        ):
+            event_list = [
+                Event(project_id=project_id, event_id=event_id)
+                for event_id in filter.event_ids
+                for project_id in filter.project_ids
+            ]
+            self.bind_nodes(event_list)
+
+            nodestore_events = [event for event in event_list if len(event.data)]
```

Comment:
Why do you need rebuild another list from the one you passed to bind_nodes ?

Reply:
You mean as opposed to mutating the original list to delete the invalid events that don't exist?

Reply:
ah ok, right, you will still have an event object for those that do not exist. Nevermind